util/network-devp2p/src/discovery.rs

```diff
@@ -450,22 +528,49 @@ impl Discovery {
 		let limit = (MAX_DATAGRAM_SIZE - 109) / 90;
 		let chunks = nearest.chunks(limit);
 		let packets = chunks.map(|c| {
-			let mut rlp = RlpStream::new_list(1);
+			let mut rlp = RlpStream::new_list(2);
 			rlp.begin_list(c.len());
 			for n in 0 .. c.len() {
 				rlp.begin_list(4);
 				c[n].endpoint.to_rlp(&mut rlp);
 				rlp.append(&c[n].id);
 			}
+			append_expiration(&mut rlp);
```

Comment:
This may be a stupid question but isn't this a modification of the protocol?
(because if so, what about compatibility?)

Reply:
No, this should affect behavior. The change is that the expiry timestamp used to be appended in `send_packet` and now it is appended to the RLP payload before calling that method.